morse_decode.rb

```diff
@@ -0,0 +1,67 @@
+MORSE_CODE = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}.freeze
+
+def decode_char(char)
+  MORSE_CODE[char]
+end
+
+def decode_word(word)
+  chars = word.chars
+  combine = ''
+  output = []
+
+  chars.each do |n|
+    if n == ' '
+      output.push(combine) unless combine.empty?
+      combine = ''
+    else
+      combine += n
+    end
+  end
+
+  output.map { |n| decode_char(n).to_s }.join
```

Comment:
- [ ] after the loop you should also push combine to the output otherwise the method will return nil everytime, this will look something like this:
```
def decode_word(word)
  chars = word.chars
  combine = ''
  output = []

  chars.each do |n|
    if n == ' '
      output.push(combine) unless combine.empty?
      combine = ''
    else
      combine += n
    end
  end
  output.push(combine) unless combine.empty?
  output.map { |n| decode_char(n).to_s }.join
end
```